### PR TITLE
feat: add button variant to pfe-autocomplete

### PR DESCRIPTION
**Pull request by eyevana**
## pfe-autocomplete

**tldr;**
The Red Hat Ecosystem Catalog (RHEC) wants to leverage pfe-autocomplete while keeping the same look and feel of its existing search bar design, which includes the search button to the right of the input field. Adding that variant to the existing pfe-autocomplete component.

### Preview

Link(s) to demo page(s) where this element can be viewed:
- [Demo](https://deploy-preview-1108--happy-galileo-ea79c4.netlify.app/elements/pfe-autocomplete/demo/) 

### What has changed and why

Link to discovery documentation (should contain all the planning and requirements for this work):
   - [Discovery](https://docs.google.com/document/d/1mXOmxgb04jqbDuRwGLTFewyYFxlFDwRmpu4qkZpb2N0/edit#)

### Testing instructions

Go the demo link above and run through the following base scenarios on the textual button example: 

1. Verify that you only see the textual button, not the icon button
2. Verify that the button is disabled until you start typing in the input field
3. Verify that clicking on an item, selects that item, and closes the drop list
4. Verify that that clicking on the clear icon clears the input and disables the button
5. Attach an event listener to pfe-autocomplete to verify that `search-event` is being triggered on click and capturing the search value. 

`document.querySelectorAll('.static')[1].addEventListener('pfe-autocomplete:search-event', (e) => console.log(e.detail.searchValue))`

#### Browser requirements

Your component should work in all of the following environments:

- [x] Latest 2 versions of Edge
- [x] Internet Explorer 11 (should be useable, not pixel perfect)
- [x] Latest 2 versions of Firefox (one on Mac OS, one of Windows OS)
- [x] Firefox 68 (or latest version for Red Hat Enterprise Linux distribution)
- [x] Latest 2 versions of Chrome (one on Mac OS, one of Windows OS)
- [x] Latest 2 versions of Safari
- [x] Android mobile device (such as the Galaxy S9)
- [x] Apple mobile device (such as the iPhone X)
- [x] Apple tablet device (such as the iPhone Pro)

### Ready-for-merge Checklist

Check off items as they are completed.  Feel free to delete items if they are not applicable.

- [ ] Expected files: all files in this pull request are related to one request or issue (no stragglers or scope-creep).
- [x] Tests have been updated to cover these changes.
- [x] Browser testing passed.
- [x] Repository compiles and tests pass.
- [x] Changelog updated (not needed for documentation updates).
- [x] Documentation (README.md, WHY.md, etc.) updated or added.
- [x] Approved by designer.

### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**

### CODE REVIEW

**Comment by eyevana** [comment]
@kylebuch8 this is ready for ya! 👀  Two things...I thought I branched off of master, but for some reason I'm seeing all these commits related to pfe-button that I need to clean up. Also, it looks like the build is failing because of the husky version I'm using? I have to take a look at that tomorrow too. I already ran `npm install` to update my lock file as recommended, but it didn't work 😞

**Comment by eyevana** [comment]
@kylebuch8 meow 🐱 have you had a chance to look into this?

**Review by kylebuch8 — APPROVED** [review]
Loading Goading Toting Moating

**Review by castastrophe — APPROVED** [review]
